dnswalk: add page (#16968)
tldr-pages/tldr: pages/common/dnswalk.md

# dnswalk

> DNS debugger.

> > "Walks" across zones and validates database consistency and best practices.

Reviewer: ```suggestion
> "Walk" across zones and validate database consistency and best practices.
```

> More information: <https://manned.org/dnswalk>.

- Debug a DNS pathway for a domain:

> `dnswalk {{domain.}}`

Reviewer 2: Shouldn't domain be a two-parter? Explain this to me

Author: Added some disambiguation in the body text to call out that it's a DNS root/FQDN.

> `dnswalk {{domain}}.`

Reviewer 3: Okay, an in the other  cases it is optional?

> `dnswalk {{domain}}.`

Reviewer 2: Oh, my bad. I forgot to suggest to do it for the whole page.

- Descend sub-domains recursively:

`dnswalk -r {{domain.}}`

> - Only perform a `dnswalk` if the zonme has been modified since the last test:

Reviewer 2: ```suggestion
- Only perform a `dnswalk` if the zonme has been [m]odified since the last test:
```

> `dnswalk -m {{domain.}}`

Reviewer 2: ```suggestion
`dnswalk -m {{domain}}.`
```

- Print [d]ebugging and status to `stderr` instead of `stdout`:

> `dnswalk -d {{domain.}}`

Reviewer 2: ```suggestion
`dnswalk -d {{domain}}.`
```

> - Suppress check for invalid characters in domain name:

Reviewer 2: ```suggestion
- Suppress check for [i]nvalid characters in domain name:
```

> `dnswalk -i {{domain.}}`

Reviewer 2: ```suggestion
`dnswalk -i {{domain}}.`
```

- Enable duplicate A record warning:

> `dnswalk -a {{domain.}}`

Reviewer 2: ```suggestion
`dnswalk -a {{domain}}.`
```

> - Enable "fascist checking" to compare A record PTR name with the forward name and report mismatches:

Reviewer 2: ```suggestion
- Enable "[F]ascist checking" to compare A record PTR name with the forward name and report mismatches:
```

> `dnswalk -F {{domain.}}`

Reviewer 2: ```suggestion
`dnswalk -F {{domain}}.`
```

> - Enable "lame delegation" to test whether the listed host is returning authoritative answers:

Reviewer 2: ```suggestion
- Enable "[l]ame delegation" to test whether the listed host is returning authoritative answers:
```

> `dnswalk -l {{domain.}}`

Reviewer 2: ```suggestion
`dnswalk -l {{domain}}.`
```
